assets/components/PersonalizeHomeModal.tsx

```diff
@@ -0,0 +1,390 @@
+                        </p>
+                    </figcaption>
+                </div>
+            </div>
+        );
+
+        const filteredTopics = (topics: Array<ITopic>) =>
+            topics.filter(({is_global}) => this.state.activeSection === '1' ? is_global != true : is_global === true);
+
+        const searchMatches = filteredTopics(wireTopics ?? [])
+            .filter((topic) => topic.label.toLowerCase().includes(this.state.searchTerm.toLowerCase()))
+            .map((topic) => (
+                <CheckboxInput
+                    key={topic._id}
+                    value={this.state.selectedTopics.includes(topic._id)}
+                    onChange={() => {
+                        this.handleChange(topic._id);
+                    }}
+                    label={gettext(topic.label)}
+                />
+            ));
+
+        const topicSearch = (searchMatches?.length ?? 0) > 0
+            ? (
+                <div style={{padding: 4}}>
+                    {searchMatches}
+                </div>
+            ) : <NoSearchMatches />;
+
+        const groupedTopics = (this.props.topics?.length ?? 0) > 0 ? (
+            <div>
+                <div className='boxed-checklist'>
+                    {filteredTopics(wireTopics).map((wireTopic) => (
+                        <CheckboxInput
+                            key={wireTopic._id}
+                            value={this.state.selectedTopics.includes(wireTopic._id)}
+                            onChange={() => {
+                                this.handleChange(wireTopic._id);
+                            }}
+                            label={gettext(wireTopic.label)}
+                        />
+                    ))}
+                </div>
+                {/* <FormToggle
+                    expanded={true}
+                    title={gettext('Agenda topics')}
+                    testId="toggle--general"
+                >
+                    {
+                        filteredTopics(agendaTopics).map((agendaTopic) => (
+                            <CheckboxInput
+                                key={agendaTopic._id}
+                                value={this.state.selectedTopics.has(agendaTopic._id)}
+                                onChange={() => {
+                                    this.handleChange(agendaTopic._id);
+                                }}
+                                label={gettext(agendaTopic.label)}
+                            />
+                        ))
+                    }
+                </FormToggle> */}
+            </div>
+        ) : (
+            NoSearchMatches()
+        );
+
+        const selectedTopics = (
+            <div className="simple-card__list pt-3">
+                {this.state.selectedTopics.map((topicId: string) => wireTopics.find((topic) => topic._id === topicId)).map((topic) => {
```

Comment:
Can you make this into a function that wouldn't return nulls and would only contain 2 iterations rather than iterating `wireTopics` as many times as there are selected topic IDs?